Uplift/Controllers/GymDetailViewController.swift

```diff
@@ -1,671 +1,225 @@
-                cell.dayLabel.text = "Sa"
-            }
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        let headerView = collectionView.dequeueReusableSupplementaryView(
+            ofKind: UICollectionView.elementKindSectionHeader,
+            withReuseIdentifier: Constants.gymDetailHeaderViewIdentifier,
+            // swiftlint:disable:next force_cast
+            for: indexPath) as! GymDetailHeaderView
+        headerView.configure(for: gym)
+        return headerView
+    }
 
-            return cell
-        } else {
-            return UITableViewCell()
-        }
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        return CGSize(width: collectionView.frame.width, height: 360)
     }
 
-    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
-        if tableView == hoursTableView {
-            let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: GymHoursHeaderView.identifier) as! GymHoursHeaderView
+    private func classesCollectionViewHeight() -> CGFloat {
+        let cellPadding: CGFloat = 12
+        let cellHeight: CGFloat = 100
+        let numberOfClasses = CGFloat(todaysClasses.count)
 
-            header.hoursLabel.text = getStringFromDailyHours(dailyGymHours: gym.gymHoursToday)
+        return numberOfClasses * cellHeight + (numberOfClasses - 1) * cellPadding
+    }
+}
 
-            let gesture = UITapGestureRecognizer(target: self, action: #selector(self.dropHours(sender:) ))
-            header.addGestureRecognizer(gesture)
+// MARK: - Layout
+extension GymDetailViewController {
+    private func setupViews() {
+        collectionView.backgroundColor = .white
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.register(GymDetailHeaderView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: Constants.gymDetailHeaderViewIdentifier)
+        collectionView.register(GymDetailHoursCell.self, forCellWithReuseIdentifier: Constants.gymDetailHoursCellIdentifier)
+        collectionView.register(GymDetailPopularTimesCell.self, forCellWithReuseIdentifier: Constants.gymDetailPopularTimesCellIdentifier)
+        collectionView.register(GymDetailFacilitiesCell.self, forCellWithReuseIdentifier: Constants.gymDetailFacilitiesCellIdentifier)
+        collectionView.register(GymDetailTodaysClassesCell.self, forCellWithReuseIdentifier: Constants.gymDetailClassesCellIdentifier)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        view.addSubview(collectionView)
 
-            return header
-        } else {
-            return nil
-        }
+        backButton.setImage(UIImage(named: "back-arrow"), for: .normal)
```

Comment:
maybe move `"back-arrow"` into a Constants file since we use it for multiple images across the app